shell.nix

```diff
@@ -1,25 +1,66 @@
-args@{ cudaVersion ? null, ...}:
-
+# This file is used by nix-shell.
+# It just takes the shell attribute from default.nix.
+{ config ? {}
+, sourcesOverride ? {}
+# If true, activates CUDA support
+, cudaSupport ? false
+# If cudaSupport is true, this needs to be set to a valid CUDA major version number, e.g. 10:
+# nix-shell --arg cudaSupport true --argstr cudaMajorVersion 10
+, cudaMajorVersion ? null
+, withHoogle ? false
+, pkgs ? import ./nix/default.nix {
+    inherit config sourcesOverride cudaSupport cudaMajorVersion;
+  }
+}:
+with pkgs;
 let
-  sources = import ./nix/sources.nix;
-  niv = (import sources.niv { }).niv;
-  # hls = (import sources.hls-nix { }).hpkgs.haskell-language-server;
-  shared = (
-    import ./shared.nix {
-      inherit (args);
-      shellBuildInputs = [
-        niv
-        # hls
-      ];
-    }
-  );
-  default = (
-    if cudaVersion == null
-    then shared.defaultCpu
-    else if cudaVersion == 9
-      then shared.defaultCuda92
-      else shared.defaultCuda102
-  );
+  # This provides a development environment that can be used with nix-shell or
+  # lorri. See https://input-output-hk.github.io/haskell.nix/user-guide/development/
+  shell = hasktorchHaskellPackages.shellFor {
+    name = "hasktorch-dev-shell";
+
+    # If shellFor local packages selection is wrong,
+    # then list all local packages then include source-repository-package that cabal complains about:
+    #packages = ps: with ps; [ ];
+
+    tools = { cabal = "3.2.0.0"; ghcide = "0.2.0"; };
+
+    # These programs will be available inside the nix-shell.
+    buildInputs =
+      with haskellPackages; [ hlint weeder ghcid lentil ]
+      # TODO: Add additional packages to the shell.
+      ++ [ ];
+
+    # Prevents cabal from choosing alternate plans, so that
+    # *all* dependencies are provided by Nix.
+    # TODO: Set to true as soon as haskell.nix issue #231 is resolved.
+    exactDeps = false;
+
+    shellHook = ''
+      export CPATH=${torch}/include/torch/csrc/api/include
+    '';
+
+    inherit withHoogle;
+  };
+
+  devops = stdenv.mkDerivation {
+    name = "devops-shell";
+    buildInputs = [
+      niv
+    ];
+    shellHook = ''
+      echo "DevOps Tools" \
+      | ${figlet}/bin/figlet -f banner -c \
+      | ${lolcat}/bin/lolcat
```

Comment:
wait is this example code :joy:

Reply:
this is taken from https://github.com/input-output-hk/cardano-skeleton/blob/master/shell.nix

here's a demo of that:
```
$ nix-shell -A devops ./shell.nix
trace: Using latest index state for hasktorch!
trace: Using index-state: 2020-06-11T00:00:00Z for hasktorch
trace: WARNING: license "GPL-2.0-or-later AND BSD-3-Clause" not found
                 ######                #######               
                  #     # ###### #    # #     # #####   ####  
                  #     # #      #    # #     # #    # #      
                  #     # #####  #    # #     # #    #  ####  
                  #     # #      #    # #     # #####       # 
                  #     # #       #  #  #     # #      #    # 
                  ######  ######   ##   ####### #       ####  
                                                              
                      #######                             
                         #     ####   ####  #       ####  
                         #    #    # #    # #      #      
                         #    #    # #    # #       ####  
                         #    #    # #    # #           # 
                         #    #    # #    # #      #    # 
                         #     ####   ####  ######  ####  
                                                          
NOTE: you may need to export GITHUB_TOKEN if you hit rate limits with niv
Commands:
  * niv update <package> - update package
```